presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/HashGenerationOptimizer.java

```diff
@@ -218,11 +221,23 @@ public PlanWithProperties visitDistinctLimit(DistinctLimitNode node, HashComputa
                     parentPreference.withHashComputation(node, hashComputation));
             VariableReferenceExpression hashVariable = child.getRequiredHashVariable(hashComputation.get());
 
+            PlanNode source = child.getNode();
+            if (node.isPartial() && node.getLimit() <= SystemSessionProperties.getHashBasedDistinctLimitThreshold(session)) {
```

Comment:
Are you planning to use `hash_based_distinct_limit_threshold = 0` to disable this feature? Should we have a separate variable to enable this?

Reply:
I think it might be cleaner to introduce a top-level property that enables approximate query results and require that it is enabled for any of the approximate optimizations including approximate results where slow workers are ignored, results produced using metadata only, this optimization, etc.

Reply:
Done

Reply:
Done